osu.Game/Rulesets/Scoring/ScoreProcessor.cs

```diff
@@ -316,6 +316,9 @@ private void revertResult(JudgementResult result)
         /// <param name="result">The <see cref="JudgementResult"/> to apply.</param>
         protected virtual void ApplyResult(JudgementResult result)
         {
+            if (HasFailed)
+                return;
+
             result.ComboAtJudgement = Combo.Value;
             result.HighestComboAtJudgement = HighestCombo.Value;
             result.HealthAtJudgement = Health.Value;
```

Comment:
Along with the previous comment, the return should be _after_ all these are set.